File: 0451-sort-characters-by-frequency/0451-sort-characters-by-frequency.java
class Solution {
    public String frequencySort(String s) {
        HashMap<Character, Integer> freq = new HashMap<>();

        for(char ch : s.toCharArray()){
            freq.put(ch, freq.getOrDefault(ch, 0)+1);
        }

        ArrayList<Map.Entry<Character, Integer>> list = new ArrayList<>(freq.entrySet());

        Collections.sort(list, (a,b)->{
            return b.getValue() - a.getValue();
        });

        StringBuilder ans = new StringBuilder();
        for(Map.Entry<Character, Integer>p : list){
            char ch = p.getKey();
            int count = p.getValue();

            for(int i=0;i<count;i++){
                ans.append(ch);
            }
        }
        return ans.toString();
    }
}

File: 0451-sort-characters-by-frequency/0451-sort-characters-by-frequency.cpp
class Solution {
public:
    string frequencySort(string s) {
        unordered_map<char, int> mpp;

        for(char ch : s){
            mpp[ch]++;
        }

        vector<pair<char,int>> v;

        for(auto x : mpp){
            v.push_back({x.first, x.second});
        }

        sort(v.begin(), v.end(), [](pair<char, int> a, pair<char, int> b){
            return a.second > b.second;
        });

        string ans = "";

        for(auto p: v){
            char ch = p.first;
            int count = p.second;

            for(int i=0;i<count;i++){
                ans+=ch;
            }
        }
        return ans;
    }
};